### PR TITLE
Enable deprecation warnings in python for development

### PR DESCRIPTION
**Pull request by peterfpeterson**
As pointed out in [this blog post](https://metarabbit.wordpress.com/2014/05/27/protip-enable-deprecation-warnings-in-python/) and [documented by python](https://docs.python.org/2/using/cmdline.html#cmdoption-w), deprecation warnings are turned off by default. This turns them back on when using the launch scripts on linux for developers.

**To test:**

Look at the new launch scripts (development and install versions) and see the difference.

*There is no associated issue.*

*This does not require release notes* because it will only be seen by developers.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by martyngigg — COMMENTED** [review]
Confirmed that we now see deprecation warnings on developer builds. This is a good step forward!

Do you know why we also get things like:
```
Not importing directory '/usr/lib/python2.7/dist-packages/plotly/graph_reference': missing __init__.py
````
which is not a `DeprecationWarning`?

What should we do about production builds for users? They are still not going to see any of our deprecation warnings.

**Comment by peterfpeterson** [comment]
The project should have a discussion about whether we enable deprecation warnings for users. I'm inclined not to and keep the normal python behavior. Having it for developers should greatly reduce the issues inside of mantid which is what I think are the bigger concern.

**Comment by martyngigg** [comment]
It just means we can never remove api methods that we've marked deprecated because we have no idea if users are using them. Agreed that this should be a wider discussion.

**Comment by martyngigg** [comment]
Do you know why we seem to get more than deprecation warnings out?

**Comment by martyngigg** [comment]
@peterfpeterson Have you had a chance to look at why we seem to get more than deprecation warnings out?

**Comment by peterfpeterson** [comment]
The [warnings module](https://docs.python.org/2/library/warnings.html) says there are ways to have more fine grained control, but doesn't say how.

**Comment by martyngigg** [comment]
`python -Wdefault::DeprecationWarning ` seems to do it for me.

**Comment by martyngigg** [comment]
That documentation is hard to follow though. I got this from `python -h`.

**Review by martyngigg — APPROVED** [review]
